tests/ui/single_field_patterns.stderr

```diff
@@ -115,5 +126,27 @@ help: try this
 LL |     let field1 = struct1.field1;
    |         ~~~~~~   ~~~~~~~~~~~~~~
 
-error: aborting due to 8 previous errors
+error: this single-variant pattern only matches one field
+  --> $DIR/single_field_patterns.rs:38:5
+   |
+LL |     let Struct { ref field1, .. } = **struct1;
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+help: try this
+   |
+LL |     let ref field1 = struct1.field1;
+   |         ~~~~~~~~~~   ~~~~~~~~~~~~~~
+
+error: this single-variant pattern only matches one field
+  --> $DIR/single_field_patterns.rs:39:5
+   |
+LL |     let Struct { ref mut field1, .. } = **struct1;
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+help: try this
+   |
+LL |     let ref mut field1 = struct1.field1;
```

Comment:
I think a better suggestion would be for the `ref` and `ref mut` pattern to borrow the field, instead of keeping it in the pattern. This is just more idiomatic Rust IMO.